src/ui/base-page-storyEdit.js

```diff
@@ -334,6 +447,40 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/main/LICENSE.
         }
     };
 
+    /**
+     * Increments or decrements a pair of counters depending on the previous and new
+     * states of a particular value. The expected possible state transitions are:
+     *
+     * - undefined to ready
+     * - ready to uploading
+     * - uploading to ready
+     * - uploading to errorReceived
+     * - errorReceived to uploading
+     *
+     *
+     * @param {String} previousState - the previous state of the value
+     * @param {String} newState - the new state of the value
+     * @param {Component} storyEdit - an instance of `sjrk.storyTelling.base.page.storyEdit`
+     */
+    sjrk.storyTelling.base.page.storyEdit.updateUploadCounters = function (previousState, newState, storyEdit) {
+        var counters = fluid.copy(storyEdit.model.uploadCounters);
+
+        if (previousState === "errorReceived") {
+            counters.errors--;
+            counters.uploads++;
```

Comment:
Is it not possible to go from "errorReceived" to "ready"?

Reply:
As we discussed at our daily check-in today, this is possible when the `resetUploadState` invoker is called, which is currently only on block deletion. I'll add a check for it in this `if` block